src/libsyntax_expand/proc_macro_server.rs

```diff
@@ -326,8 +326,20 @@ impl Ident {
             false
         }
     }
-    fn new(sym: Symbol, is_raw: bool, span: Span) -> Ident {
+    fn nfc_normalize(sym: Symbol) -> (Symbol, SymbolStr) {
```

Comment:
Could you avoid duplicating the code between here and lexer?

```rust
fn nfc_normalize(string: &str) -> Symbol { ... }
```
defined in the lexer will do the job.

(It won't be entirely optimal due to re-interning, but that's not a problem until profiling proves otherwise.)

Reply:
Done.